Update the Go Quickstart README (#1896)
minio/minio-go: README.md

# MinIO Go Client SDK for Amazon S3 Compatible Cloud Storage [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io) [![Sourcegraph](https://sourcegraph.com/github.com/minio/minio-go/-/badge.svg)](https://sourcegraph.com/github.com/minio/minio-go?badge) [![Apache V2 License](https://img.shields.io/badge/license-Apache%20V2-blue.svg)](https://github.com/minio/minio-go/blob/master/LICENSE)

> The MinIO Go Client SDK provides APIs to access any Amazon S3 compatible object storage.

Reviewer: ```suggestion
The MinIO Go Client SDK provides simple APIs to access any Amazon S3 compatible object storage.
```

Simple is on purpose in competing SDK AWS SDK Go is horrendous.

Reviewer 2: What about `straightforward` rather than `simple`.

`simple` often carries a connotation that it should be easy, even if you don't have the requisite back knowledge. We try to avoid it in technical writing because it can often overpromise with how people read it.

Author: What @djwfyi said about _simple_ is exactly why I removed it. Straightforward is a good alternative. 👍🏻

> This Quickstart Guide covers how to install the MinIO client SDK, connect to MinIO, and a walkthrough for a basic file uploader.

Reviewer 2: Let's make this one parallel:

```suggestion
This Quickstart Guide covers how to install the MinIO client SDK, connect to MinIO, and create a sample basic file uploader.
```

For a complete list of APIs and examples, see the [godoc documentation](https://pkg.go.dev/github.com/minio/minio-go/v7) or [Go Client API Reference](https://min.io/docs/minio/linux/developers/go/API.html).

These examples presume a working [Go development environment](https://golang.org/doc/install) and the [MinIO `mc` command line tool](https://min.io/docs/minio/linux/reference/minio-mc.html).

## Download from Github

```sh

> go install github.com/minio/minio-go/v7@latest

Author: This is what worked for me, is `@latest` appropriate here?

Reviewer: No libraries do `go get` not `go install` - `go install` is meant for binaries.

Author: @harshavardhana is there more to know about how to use `go get` for the Go SDK? When I tried the original command, I got an error that said to use `go install`.
```
go-dev[feorlen]:~$ go get github.com/minio/minio-go/v7
go: go.mod file not found in current directory or any parent directory.
	'go get' is no longer supported outside a module.
	To build and install a command, use 'go install' with a version,
	like 'go install example.com/cmd@latest'
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
```
```
go-dev[feorlen]:~/dev$ go install github.com/minio/minio-go/v7
go: 'go install' requires a version when current directory is not in a module
	Try 'go install github.com/minio/minio-go/v7@latest' to install the latest version
```

Reviewer: You need a project folder first i.e 

```
mkdir myproject
go mod init
go get github.com/minio/minio-go/v7@latest
.. then you code
```

Reviewer: so if you are a Go developer you would know why this SDK says `go get`

Reviewer: If you read the error message 

> go: go.mod file not found in current directory or any parent directory.

That is generally explaining what it is expecting when you do `go get`

Author: Ah ok so my issue is order of operations. 👍🏻

> go install github.com/minio/minio-go/v7@latest

Reviewer: ```suggestion
go get github.com/minio/minio-go/v7@latest
```

}
```

> **1. Create a test file containing data:**

Author: Using bold formatting here because H4 is a bit too much nesting.

> On Linux systems, you may do this with `dd`:

Author: Can anybody recommend an equivalent command for Windows? 

`/dev/urandom` exists on Mac although specific OS version and user permissions might cause trouble?

Reviewer 2: ```
fsutil file createnew <file name/path/extension> <size of file in bytes>
```

```
fsutil file createnew "C:\Users\<username>\Desktop\sample.txt" 104857600
```

Author: > fsutil file createnew ...

Thank you @djwfyi !!

```
dd if=/dev/urandom of=/tmp/testdata bs=2048 count=10
```

> **2. Run FileUploader with the following commands:**

Author: Feedback on these commands very welcome, this is what worked for me. I'm not sure the usual Go way of referring to these operations.

For now, this is not intended to be a comprehensive discussion. Just the basics for someone generally familiar with the language and tools.

```sh
go mod init example/FileUploader
go get github.com/minio/minio-go/v7
go get github.com/minio/minio-go/v7/pkg/credentials
go run FileUploader.go
```

> The output resembles the following:

Reviewer 2: Not vital, but I'd suggest adding a blank line before and after this one for easier reading when editing this file in the future.

Author: Oh yes, you are correct. 👍🏻

> ```

Reviewer 2: ```suggestion
```sh
```

Author: That and several others. Yup.

* [put-encrypted-object.go](https://github.com/minio/minio-go/blob/master/examples/s3/put-encrypted-object.go)
* [get-encrypted-object.go](https://github.com/minio/minio-go/blob/master/examples/s3/get-encrypted-object.go)
* [fput-encrypted-object.go](https://github.com/minio/minio-go/blob/master/examples/s3/fputencrypted-object.go)

### Full Examples : Presigned Operations

* [presignedgetobject.go](https://github.com/minio/minio-go/blob/master/examples/s3/presignedgetobject.go)
* [presignedputobject.go](https://github.com/minio/minio-go/blob/master/examples/s3/presignedputobject.go)
* [presignedheadobject.go](https://github.com/minio/minio-go/blob/master/examples/s3/presignedheadobject.go)
* [presignedpostpolicy.go](https://github.com/minio/minio-go/blob/master/examples/s3/presignedpostpolicy.go)

## Explore Further

* [Godoc Documentation](https://pkg.go.dev/github.com/minio/minio-go/v7)
* [Complete Documentation](https://min.io/docs/minio/kubernetes/upstream/index.html)
* [MinIO Go Client SDK API Reference](https://min.io/docs/minio/linux/developers/go/API.html)

## Contribute

[Contributors Guide](https://github.com/minio/minio-go/blob/master/CONTRIBUTING.md)

## License

> This SDK is distributed under the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0), see [LICENSE](https://github.com/minio/minio-go/blob/master/LICENSE) and [NOTICE](https://github.com/minio/minio-go/blob/master/NOTICE) for more information.

Author: Is this license information still accurate?

Reviewer: SDKs are all Apache 2.0
